.github/workflows/build.yaml

```diff
@@ -28,12 +28,7 @@ jobs:
           arguments: |
             --build-cache
             build
-            publishToMavenLocal
```

Comment:
What I though about, when proposing to put things into separate jobs that run in parallel, was to still do the publishing here and then put the `~.m2` folder into a job artifact that's used by the downstream jobs. The changes right now result in building larger parts of the core three times due to the inefficient build cache.

Reply:
I'll keep it as it is for now, since it's still faster than the current solution.
Building 3 times shouldn't be that expensive once we get the cache working in a followup ticket.

Reply:
The only thing is that this script executed this way might not pick up the cache.